contracts/contracts/RequestManager.sol

```diff
@@ -391,4 +393,9 @@ contract RequestManager is Ownable {
         gasPrice = newGasPrice;
         serviceFeePPM = newServiceFeePPM;
     }
+
+    function addTargetChain(uint256 targetChainId, uint256 finalizationTime) external onlyOwner {
+        require(finalizationTime > 0, "Finalization time of roll up must be greater 0");
+        finalizationTimes[targetChainId] = finalizationTime;
+    }
```

Comment:
As this function can be called multiple times for the same chain ID, I would rename it to something like `setFinalizationTime`.